lib/migrate.js

```diff
@@ -96,8 +96,9 @@ function mixinMigrate(CouchDB) {
       self.getModifyIndexes(mo, indexes, isMigrate, function(err, results) {
         debug('start drop and add indexes %j for model %j', results, model);
         if (err) return cb(err);
-        assert(typeof results === 'object',
+        assertMsg = helpers.doAssert(typeof results === 'object',
```

Comment:
I'm not really sure what the point of this is anymore; if the helper function isn't taking in callbacks, then we can just have logical boolean checks in their appropriate locations, without needing to throw, catch, and then return a boolean result. Otherwise, we'll want to wrap any calls to functions that use assert in async areas with try-catch.